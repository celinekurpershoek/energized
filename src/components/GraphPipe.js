import React from "react";
import { DifferenceBlock } from "./DifferenceBlock";

export const GraphPipe = ({
  startValue,
  heightValue,
  difference,
  percentage
}) => (
  <div className="graph__pipes">
    <span
      title={`${startValue} KWH`}
      className="graph__pipe graph__pipe--vertical"
      style={{
        height: `${heightValue}%`
      }}
    />
    <DifferenceBlock difference={difference} percentage={percentage} />
  </div>
);
