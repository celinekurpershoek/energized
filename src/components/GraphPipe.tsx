import * as React from "react";
import DifferenceBlock from "./differenceBlock";

interface IProps {
  startValue: number;
  heightValue: number;
  difference: number;
  percentage: number;
}

export default function GraphPipe(props: IProps) {
  const { startValue, heightValue, difference, percentage } = props;

  return (
    <div className="graph__pipes">
      <span
        title={`${startValue} KWH`}
        className="graph__pipe graph__pipe--vertical"
        style={{
          height: `${heightValue}%`
        }}
      />
      {percentage >= 1 && (
        <DifferenceBlock difference={difference} percentage={percentage} />
      )}
    </div>
  );
}
