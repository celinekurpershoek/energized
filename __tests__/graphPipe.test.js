import React from "react";
import renderer from "react-test-renderer";
import GraphPipe from "../src/components/GraphPipe";

it("Correctly render GraphPipe component", () => {
  const component = renderer.create(
    <GraphPipe startValue="0" heightValue="10" difference="5" percentage="50" />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
