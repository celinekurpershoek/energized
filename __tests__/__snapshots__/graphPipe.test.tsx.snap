// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Correctly render GraphPipe component 1`] = `
<div
  className="graph__pipes"
>
  <span
    className="graph__pipe graph__pipe--vertical"
    style={
      Object {
        "height": "10%",
      }
    }
    title="0 KWH"
  />
  <span
    className="graph__pipe graph__pipe--vertical graph__pipe--difference"
    style={
      Object {
        "height": "50%",
      }
    }
    title="5KWH (+50%)"
  />
</div>
`;
